mitiq/interface/mitiq_qibo/conversions.py

```diff
@@ -0,0 +1,529 @@
+            else:
+                registers[register] = {idx: qubits[qubit]}
+                gate_list.append(("M", register, None))
+
+        else:
+            pieces = [x for x in re.split("[()]", command) if x]
+            if len(pieces) == 1:
+                gatename, params = pieces[0], None
+                if gatename not in gates.QASM_GATES:
+                    if gatename != "sx" and gatename != "sxdg": 
+                        raise_error(
+                            ValueError,
+                            "QASM command {} is not recognized." "".format(command),
+                        )
+                if gatename in gates.PARAMETRIZED_GATES:
+                    raise_error(
+                        ValueError,
+                        "Missing parameters for QASM " "gate {}.".format(gatename),
+                    )
+
+            elif len(pieces) == 2:
+                gatename = pieces[0]
+                params = pieces[1].replace(" ", "").split(",")
+                if gatename not in gates.PARAMETRIZED_GATES:
+                    raise_error(
+                        ValueError, "Invalid QASM command {}." "".format(command)
+                    )
+                try:
+                    for i, p in enumerate(params):
+                        denominator = 1
+                        if "/" in p:
+                            p, denominator = p.split("/")
+                        if "pi" in p:
+                            from functools import reduce
+                            from operator import mul
+
+                            s = p.replace("pi", str(pi)).split("*")
+                            p = reduce(mul, [float(j) for j in s], 1)
+                        params[i] = float(p)
+                except ValueError:
+                    raise_error(
+                        ValueError,
+                        "Invalid value {} for gate parameters." "".format(params),
+                    )
+
+            else:
+                raise_error(
+                    ValueError,
+                    "QASM command {} is not recognized." "".format(command),
+                )
+
+            # Add gate to gate list
+            qubit_list = []
+            for qubit in read_args(args):
+                if qubit not in qubits:
+                    raise_error(
+                        ValueError,
+                        "Qubit {} is not defined in QASM " "code.".format(qubit),
+                    )
+                qubit_list.append(qubits[qubit])
+            if gatename == "sx": 
+                gate_list.append(("RX", list(qubit_list),  [pi/2]))
+            elif gatename == "sxdg": 
+                gate_list.append(("RX", list(qubit_list),  [-pi/2]))
+            else: 
+                gate_list.append((gates.QASM_GATES[gatename], list(qubit_list), params))
+
+    # Create measurement gate qubit lists from registers
+    for i, (gatename, register, _) in enumerate(gate_list):
```

Comment:
Seems that `register` here is a `List[int]` but seems we want a string to index `registers` in line 524.

Reply:
I've realised that in the case of having a measurement instead of an usual gate register is indeed a string. I've changed the order in wich we assign register to gate list at first becuase I was assigning register to the second index when it should be third. Then, the third gate_list index can be not only List[Float] or None but also str (when having a measurment).